controllers/baseboardmanagement_controller.go

```diff
@@ -165,6 +175,7 @@ func (r *BaseboardManagementReconciler) reconcilePower(ctx context.Context, bm *
 	// Setting baseboard management to desired power state
 	_, err = bmcClient.SetPowerState(ctx, string(bm.Spec.Power))
 	if err != nil {
+		r.recorder.Event(bm, corev1.EventTypeWarning, "SetPowerState", fmt.Sprintf("failed to set power state: %v", err))
```

Comment:
I think we can rejig the warnings a little. 

```suggestion
		r.recorder.Eventf(bm, corev1.EventTypeWarning, EventSetPowerStateFailed, "reconciling power: %v", err)
```

Then define some event constants.

```go
const (
  EventGetPowerStateFailed = "GetPowerStateFailed"
  EventSetPowerStateFailed = "SetPowerStateFailed"
)
```